### DISCUSSION TITLE
"extends" property in Nitro?

### DISCUSSION THREAD

**Discussion by justindasilva** [nitrojs/nitro#550] [Ideas]
I really like the new "extends" property offered in Nuxt 3 (https://v3.nuxtjs.org/examples/advanced/config-extends/)

Would there be any plans to support something similar in Nitro?

Our team has a use case where we want to create a simple application with some default routes, default ESLint/typescript configuration, etc.. Other teams will be utilizing the defaults we created but then extending it with more functionality on top.

Supporting an "extends" in Nitro would actually make things really nice for us :)

**Comment by qvdp** [comment]
Hi, is there any workaround to achieve this? I would like to be able, depending on the context, to mount another nitro instance when starting my primary instance.

More specifically, I'm trying to implement a flexible architecture where I can dynamically choose between two deployment modes for a child Nitro application:

- Mode 1: Standalone server for server runtime - Launch a separate Nitro child server on a different port (e.g., port 4000) that runs independently alongside the main server, but I am not sure this is suitable for production purpose.
- Mode 2: Embedded routing for edge runtime - Mount the child Nitro instance directly into the parent server under a specific route prefix (e.g., ⁠/_proxy/**) so everything runs on the same runtime.

I'm currently working with a Nitro plugin approach like this:
```
 export default defineNitroPlugin(async (nitroApp) => {
  const nitro = await createNitro({
    dev: process.dev,
    // child config here
  });

  if (process.env.CHILD_INSTANCE_MODE === 'standalone') {
    const server = createDevServer(nitro);
    await server.listen(4000);
    return;
  }
  
  // This is where I'm stuck - trying to mount nitro instance on /_unbase
  nitroApp.router.use('/_proxy', nitro); // nitro doesn't expose h3App directly
});
```
 The standalone mode works fine, but I'm having trouble with the embedded mode because ⁠createNitro() doesn't directly expose an h3App that I could mount as middleware.

Has anyone found a way to programmatically mount one Nitro instance inside another ?
